ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/services/TestDefaultKsqlClientFactory.java

```diff
@@ -11,8 +13,9 @@
 public class TestDefaultKsqlClientFactory {
 
   // Creates an instance with no auth
-  public static SimpleKsqlClient instance(Map<String, Object> clientProps) {
-    return new DefaultKsqlClient(Optional.empty(), clientProps);
+  public static SimpleKsqlClient instance(Map<String, Object> clientProps,
+      final BiFunction<Integer, String, SocketAddress> socketAddressFactory) {
```

Comment:
Is it worth having two versions: one with and one without this function?  Seems like most things won't need the ability to override the host.